test/features/DebugSession.test.ts

```diff
@@ -0,0 +1,491 @@
+                }),
+            });
+            const debugSession = stubInterface<DebugSession>({
+                configuration: stubInterface<DebugConfiguration>({
+                    createTemporaryIntegratedConsole: false
+                })
+            });
+
+            const debugAdapterDescriptor = await debugSessionFeature.createDebugAdapterDescriptor(debugSession, undefined);
+
+            // Confirm debugAdapterDescriptor is of type debugadapternamedpipeserver
+            assert.ok(debugAdapterDescriptor instanceof DebugAdapterNamedPipeServer);
+            assert.equal(debugAdapterDescriptor.path, "testPipeName");
+        });
+    });
+});
+
+describe("DebugSessionFeature E2E", function slowTests() {
+    this.slow(20000); // Will warn if test takes longer than 10s and show red if longer than 20s
+
+    if (process.platform == "darwin") {
+        this.timeout(60000); // The MacOS test runner is sloooow in Azure Devops
+    }
+    before(async () => {
+        // Registers and warms up the debug adapter and the PowerShell Extension Terminal
+        await ensureEditorServicesIsConnected();
+    });
+
+    it("Starts and stops a debugging session", async () => {
+        // Inspect the debug session via the started events to ensure it is correct
+        let startDebugSession: DebugSession;
+        let stopDebugSession: DebugSession;
+        const interactiveSessionConfig = defaultDebugConfigurations[DebugConfig.InteractiveSession];
+        // Asserts dont seem to fire in this event or the event doesnt resolve in the test code flow, so we need to "extract" the values for later use by the asserts
+
+        const startDebugEvent = debug.onDidStartDebugSession((newDebugSession) => {
+            startDebugEvent.dispose();
+            startDebugSession = newDebugSession;
+            const stopDebugEvent = debug.onDidTerminateDebugSession((terminatedDebugSession) => {
+                stopDebugEvent.dispose();
+                stopDebugSession = terminatedDebugSession;
+            });
+        });
+
+        const debugSessionStarted = await debug.startDebugging(undefined, interactiveSessionConfig);
+        assert.ok(debugSessionStarted, "Debug session should start");
+        assert.equal(startDebugSession!.name, interactiveSessionConfig.name, "Debug session name should match when started");
+        // debugSession var should be populated from the event before startDebugging completes
+        await debug.stopDebugging(startDebugSession!);
+
+        assert.equal(stopDebugSession!.name, interactiveSessionConfig.name, "Debug session name should match when stopped");
+        assert.equal(stopDebugSession!.configuration.internalConsoleOptions, "neverOpen", "Debug session should always have neverOpen internalConsoleOptions");
+        assert.ok(stopDebugSession!, "Debug session should stop");
+    });
+
+    describe("Binary Modules", () => {
+        before(async () => {
+            await InstallCSharpExtension();
```

Comment:
Ah, found where that happens.

Reply:
Yeah it's convoluted logic and I thought it messed up reading how the test works too much so I extracted it out. We could do this just as a pre-requisite but I figured it made sense to be on-demand so if you didn't want to run the binary module tests or were testing otherwise, you aren't slowed down waiting for the extension to install. If we ever implement caching in azure devops, this can also just be cached making it a moot point.